kroxylicious-integration-tests/src/test/java/io/kroxylicious/proxy/filter/validation/JsonSyntaxRecordValidationIT.java

```diff
@@ -5,229 +5,233 @@
-                        .build());
+                .addToFilters(createFilterDef(true, topic1));
 
         try (var tester = kroxyliciousTester(config);
                 var producer = tester.producer(Map.of(LINGER_MS_CONFIG, 5000, TRANSACTIONAL_ID_CONFIG, randomUUID().toString()))) {
             producer.initTransactions();
             producer.beginTransaction();
-            Future<RecordMetadata> invalid = producer.send(new ProducerRecord<>(TOPIC_1, "my-key", SYNTACTICALLY_INCORRECT_JSON));
-            Future<RecordMetadata> valid = producer.send(new ProducerRecord<>(TOPIC_2, "my-key", SYNTACTICALLY_CORRECT_JSON));
+            var invalid = producer.send(new ProducerRecord<>(topic1.name(), "my-key", SYNTACTICALLY_INCORRECT_JSON));
+            var valid = producer.send(new ProducerRecord<>(topic2.name(), "my-key", SYNTACTICALLY_CORRECT_JSON));
             producer.flush();
-            assertInvalidRecordExceptionThrown(invalid, "value was not syntactically correct JSON");
-            assertInvalidRecordExceptionThrown(valid, "Invalid record in another topic-partition caused whole ProduceRequest to be invalidated");
+            assertThatFutureFails(invalid, InvalidRecordException.class, "value was not syntactically correct JSON");
+            assertThatFutureFails(valid, InvalidRecordException.class, "Invalid record in another topic-partition caused whole ProduceRequest to be invalidated");
             producer.abortTransaction();
         }
     }
 
     @Test
-    void testPartiallyInvalidJsonNotConfiguredToForwardAllRejected(KafkaCluster cluster, Admin admin) throws Exception {
+    void partiallyInvalidJsonNotConfiguredToForwardAllRejected(KafkaCluster cluster, Topic topic1, Topic topic2) {
         assertThat(cluster.getNumOfBrokers()).isOne();
-        createTopics(admin, new NewTopic(TOPIC_1, 1, (short) 1), new NewTopic(TOPIC_2, 1, (short) 1));
 
-        boolean forwardPartialRequests = false;
         var config = proxy(cluster)
-                .addToFilters(new FilterDefinitionBuilder(RecordValidation.class.getName())
-                        .withConfig("forwardPartialRequests", forwardPartialRequests, "rules",
-                                List.of(Map.of("topicNames", List.of(TOPIC_1, TOPIC_2), "valueRule",
-                                        Map.of("allowsNulls", true, "syntacticallyCorrectJson", Map.of("validateObjectKeysUnique", true)))))
-                        .build());
+                .addToFilters(createFilterDef(false, topic1));
 
         try (var tester = kroxyliciousTester(config);
-                var producer = getProducer(tester, 5000, 16384)) {
-            Future<RecordMetadata> invalid = producer.send(new ProducerRecord<>(TOPIC_1, "my-key", SYNTACTICALLY_INCORRECT_JSON));
-            Future<RecordMetadata> valid = producer.send(new ProducerRecord<>(TOPIC_2, "my-key", SYNTACTICALLY_CORRECT_JSON));
+                var producer = tester.producer(Map.of(LINGER_MS_CONFIG, 5000))) {
```

Comment:
Do we actually need `linger.ms` in this test? (I only ask as I think its been removed from others)

Reply:
I think so, we need `linger.ms` to provoke batching so they go out in the same produce request. Maybe it shouldn't be associate with JsonSyntax since it's a general behaviour, we could test it against a filter set up to block null/empty in a test like `PartialForwardingRecordValidationIT.java` (what's a good name for this feature)

Reply:
or maybe it's good to have at least one test checking it composes with a specific validation in addition to a separate test.

Same with the null/empty tests, they can be used on their own without any schema/syntax involvement

Reply:
>Maybe it shouldn't be associate with JsonSyntax since it's a general behaviour

Yeah, thats a very strong argument for a separate test. 

>  we could test it against a filter set up to block null/empty in a test like PartialForwardingRecordValidationIT.java (what's a good name for this feature)

I'm not sure which bit you want a name for. The null/empty validation? Or to rename `PartialForwarding`?

Reply:
I'm going to leave this for another day.